src/Faker/Provider/es_MX/Person.php

```diff
@@ -0,0 +1,454 @@
+
+
+    public function lastName()
+    {
+        return static::randomElement(static::$lastName) . ' ' . static::randomElement(static::$lastName);
+    }
+
+    public function lastNameMother()
+    {
+        return static::randomElement(static::$lastNames);
+    }
+
+    public function lastNameFather()
+    {
+        return static::randomElement(static::$lastNames);
+    }
+    /**
+     * generates CURP (uique identity code for both citizens and residents of Mexico)
+     *
+     * @param      string  $firstName       First name
+     * @param      string  $lastNameFather  Fathers last name
+     * @param      string  $lastNameMother  Mothers last name
+     * @param      \DateTime  $birthDate    Birth date
+     * @param      string  $gender          Gender
+     * @param      string  $state           State of birth
+     *
+     * @return     string  generated curp
+     *
+     * @link       https://github.com/hectorip/RFC-CURP-Mexico/blob/master/src/mxk.js ported from this
+     * @link       https://en.wikipedia.org/wiki/Unique_Population_Registry_Code
+     */
+    public static function curp($firstName = null, $lastNameFather = null, $lastNameMother = null, $birthDate = null, $gender = null, $state = null)
+    {
+        $gender = in_array($gender, array(Person::GENDER_MALE, Person::GENDER_FEMALE)) ? $gender : static::randomElement(array(Person::GENDER_MALE, Person::GENDER_FEMALE));
+
+        if ($gender === Person::GENDER_MALE) {
+            $firstName = self::removeAccents(self::removeCommonNames(mb_strtoupper($firstName ? $firstName : static::firstNameMale())));
+        } else {
+            $firstName = self::removeAccents(self::removeCommonNames(mb_strtoupper($firstName  ?$firstName : static::firstNameFemale())));
+        }
+
+        $lastNameFather = self::removeAccents(self::removePrefixes(mb_strtoupper($lastNameFather ? $lastNameFather : static::randomElement(static::$lastNames))));
+        $lastNameMother = self::removeAccents(self::removePrefixes(mb_strtoupper($lastNameMother ? $lastNameMother : static::randomElement(static::$lastNames))));
+        $birthDate = $birthDate ? $birthDate:\Faker\Provider\DateTime::dateTimeBetween();
```

Comment:
no it is not

if $birthDate exists then pass it to $birthDate, otherwise take random DateTime::dateTimeBetween()